python_modules/dagster/dagster/core/object_store.py

```diff
@@ -84,18 +84,62 @@ def has_object(self, context, paths):
     def rm_object(self, context, paths):
         pass
 
+    def _check_for_unsupported_composite_overrides(self, runtime_type):
+        composite_overrides = {
```

Comment:
This expression seems too clever by half.

What do you think of

```py
any(for t in runtime_type.inner_types if t.name in self.TYPE_STORAGE_PLUGIN_REGISTRY)
```
along with a comment? an explicit check on ```runtime_type.is_list or runtime_type.is_nullable``` could also be clarifying

Reply:
is it the use of the set comprehension that seems unclear?

Reply:
I think the most confusing part is that the replication of ```t.name in self.TYPE_STORAGE_PLUGIN_REGISTRY``` as the value in the set. It looks like you are trying to build a meaningful set of values at first blush but you're aren't really using the properties of the set (the logic would be the same with a list). The bools are meaningless.

I think any does express what you are trying to express more explicitly.